src/doc/trpl/guessing-game.md

```diff
@@ -0,0 +1,991 @@
+
+Nice! You can see I even added spaces before my guess, and it still figured
+out that I guessed 76. Run the program a few times, and verify that guessing
+the number works, as well as guessing a number too small.
+
+Now we’ve got most of the game working, but we can only make one guess. Let’s
+change that by adding loops!
+
+# Looping
+
+The `loop` keyword gives us an infinite loop. Let’s add that in:
+
+```rust,no_run
+extern crate rand;
+
+use std::io;
+use std::cmp::Ordering;
+use rand::Rng;
+
+fn main() {
+    println!("Guess the number!");
+
+    let secret_number = rand::thread_rng().gen_range(1, 101);
+
+    println!("The secret number is: {}", secret_number);
+
+    loop {
+        println!("Please input your guess.");
+
+        let mut guess = String::new();
+
+        io::stdin().read_line(&mut guess)
+            .ok()
+            .expect("failed to read line");
+
+        let guess: u32 = guess.trim().parse()
+            .ok()
+            .expect("Please type a number!");
+
+        println!("You guessed: {}", guess);
+
+        match guess.cmp(&secret_number) {
+            Ordering::Less    => println!("Too small!"),
+            Ordering::Greater => println!("Too big!"),
+            Ordering::Equal   => println!("You win!"),
+        }
+    }
+}
+```
+
+And try it out. But wait, didn’t we just add an infinite loop? Yup. Remember
+our discussion about `parse()`? If we give a non-number answer, we’ll `return`
+and quit. Observe:
+
+```bash
+$ cargo run
+   Compiling guessing_game v0.0.1 (file:///home/you/projects/guessing_game)
+     Running `target/guessing_game`
+Guess the number!
+The secret number is: 59
+Please input your guess.
+45
+You guessed: 45
+Too small!
+Please input your guess.
+60
+You guessed: 60
+Too big!
+Please input your guess.
+59
+You guessed: 59
+You win!
+Please input your guess.
+quit
+thread '<main>' panicked at 'Please type a number!'
+```
+
+Ha! `quit` actually quits. As does any other non-number input. Well, this is
+suboptimal to say the least. First, let’s actually quit when you win the game:
+
+```rust,no_run
+extern crate rand;
+
+use std::io;
+use std::cmp::Ordering;
+use rand::Rng;
+
+fn main() {
+    println!("Guess the number!");
+
+    let secret_number = rand::thread_rng().gen_range(1, 101);
+
+    println!("The secret number is: {}", secret_number);
+
+    loop {
+        println!("Please input your guess.");
+
+        let mut guess = String::new();
+
+        io::stdin().read_line(&mut guess)
+            .ok()
+            .expect("failed to read line");
+
+        let guess: u32 = guess.trim().parse()
+            .ok()
+            .expect("Please type a number!");
+
+        println!("You guessed: {}", guess);
+
+        match guess.cmp(&secret_number) {
+            Ordering::Less    => println!("Too small!"),
+            Ordering::Greater => println!("Too big!"),
+            Ordering::Equal   => {
+                println!("You win!"),
+                return;
+            }
```

Comment:
I think this branch can technically be `return println!("You win!")`, but that may be a bit overkill for this early on perhaps.